js/src/components/paid-ads/audience-section.js

```diff
@@ -2,25 +2,71 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { SelectControl } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
+import useCountryKeyNameMap from '.~/hooks/useCountryKeyNameMap';
 import Section from '.~/wcdl/section';
 import AudienceCountrySelect from '.~/components/audience-country-select';
 
+function toCountryOptions( countryCodes, countryNameMap ) {
+	return countryCodes.map( ( code ) => ( {
+		label: countryNameMap[ code ],
+		value: code,
+	} ) );
+}
+
+/**
+ * Renders <Section> and <Section.Card> UI with country(s) selector.
+ *
+ * @param {Object} props React props.
+ * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @param {boolean} [props.multiple=true] Whether the selector is multi-selected.
+ * @param {boolean} [props.disabled=false] Whether the selector is disabled.
+ * @param {JSX.Element} [props.countrySelectHelperText] Helper text to be displayed under the selector.
+ */
 const AudienceSection = ( props ) => {
 	const {
 		formProps: { getInputProps },
+		multiple = true,
 		disabled = false,
 		countrySelectHelperText,
 	} = props;
 
+	const countryNameMap = useCountryKeyNameMap();
 	const inputProps = getInputProps( 'country' );
+
+	/**
+	 * Hack to prevent tags being removed even when disabled.
+	 * This could be removed after replacing <AudienceCountrySelect> with <TreeSelectControl>.
+	 */
 	if ( disabled ) {
 		inputProps.onChange = undefined;
 	}
```

Comment:
I had [removed this](https://github.com/woocommerce/google-listings-and-ads/commit/26ac0d37e9add0f9dfa8a1de8e9d9f700c924acc#diff-5c6f0d2e75d046351c001bcf2b5de4756de5193bae214db825e21d17b9597c05L20-L22) while developing. And then I recalled `@woocommerce/components` is a runtime version injected via DEWP. I worried it might break again someday if only has the CSS workaround, so I added this back to have a second layer of defense to secure the wanted disabled function.

I will open an issue to upstream repo. Hope we can remove this workaround soon.

Reply:
That's a careful thinking and a totally valid point! Agree 😃 👍